### PR TITLE
Pass edition to rustdoc.

### LINKED ISSUE #5279: Pass `--edition` to rustdoc when an edition is configured
In https://github.com/rust-lang/rust/pull/49451 i added an `--edition` flag to rustdoc that made it compile crates and doctests in the given edition. Cargo should be extended to call this flag when an edition is present in a project's Cargo.toml.

At present the flag is unstable, so a `-Z unstable-options` will need to be passed to rustdoc for it to work properly. I'm not sure how cargo handles the flag for rustc, but i imagine something similar should happen for rustdoc.

### PR DESCRIPTION
**Pull request by ehuss**
Fixes #5279.

### CODE REVIEW

File: tests/testsuite/doc.rs [hunk 1]
```diff
@@ -1464,3 +1464,33 @@ fn doc_workspace_open_help_message() {
...
+            version = "0.0.1"
+            authors = []
+            rust = "2018"
+        "#,
+        )
+        .file("src/lib.rs", "")
+        .build();
+
+    assert_that(
+        p.cargo("doc").arg("-v").masquerade_as_nightly_cargo(),
```
Comment (matklad) [inline tests/testsuite/doc.rs]: You can write this compactly as `p.cargo("doc -v").masquerade_as_nightly_cargo()`, `cargo` function does simple argument splitting.

**Comment by matklad** [comment]
@bors r+ 

Thanks!

**Comment by matklad** [comment]
Test failure seems legit. Perhaps the nightly at travis is too old?

**Comment by ehuss** [comment]
> Test failure seems legit. Perhaps the nightly at travis is too old?

Indeed it is too old.  Perhaps now that rust-lang/rust#49033 is fixed we can switch back to regular "nightly" channel?

**Comment by matklad** [comment]
@ehuss probably! I have no time to actually investigate this, could you perhaps just try to change CI configuration in this PR?

**Comment by ehuss** [comment]
@matklad Looks like everything is passing now.  Let me know if you want the CI changes in a separate PR, or if it's OK to ride on this one.

**Comment by matklad** [comment]
@bors r+

:)

**Pull request merged**